model_compression_toolkit/gptq/pytorch/quantizer/base_pytorch_gptq_quantizer.py

```diff
@@ -0,0 +1,103 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from abc import abstractmethod
+from typing import Union, Dict, List
+
+from torch import Tensor
+
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.core.common.constants import FOUND_TORCH
+from model_compression_toolkit.gptq.common.gptq_constants import WEIGHTS_QUANTIZATION_PARAMS
+
+from model_compression_toolkit.quantizers_infrastructure import TrainableQuantizerWeightsConfig, \
+    TrainableQuantizerActivationConfig, BasePytorchTrainableQuantizer
+
+if FOUND_TORCH:
+
+    class BasePytorchGPTQTrainableQuantizer(BasePytorchTrainableQuantizer):
+        """
+        A base class for trainable Pytorch quantizer for GPTQ.
+        """
+
+        def __init__(self,
+                     quantization_config: Union[TrainableQuantizerWeightsConfig, TrainableQuantizerActivationConfig]):
+            """
+            Initializes BasePytorchGPTQTrainableQuantizer object.
+
+            Args:
+                quantization_config: quantizer config class contains all the information about a quantizer configuration.
+            """
+
+            super().__init__(quantization_config)
+
+        def update_layer_quantization_params(self, layer) -> (Dict[str, Tensor], Dict[str, Dict], Dict):
+            """
+            A Function to calculate the needed change in attributes in NodeQuantizationConfig after retraining.
+
+            Returns:
+                3 dictionaries describing the change in layer's weights, weights config, activation config
+                that changed during GPTQ retraining.
+                Keys must match NodeQuantizationConfig attributes
+
+            """
+            weights = {}
+            for weight, quantizer_vars, quantizer in layer.get_weights_vars():
+                weights.update({weight: quantizer(training=False, inputs=quantizer_vars)})
+
+            quant_config = {WEIGHTS_QUANTIZATION_PARAMS: self.get_quant_config()}
+
+            return weights, quant_config, {}
+
+        def get_aux_variable(self) -> List[Tensor]:
+            """
+            This function return a list with the quantizer's quantization auxiliary variables.
+
+            Returns: A list with the quantization auxiliary variables.
+
+            """
+
+            return []
+
+        def get_quantization_variable(self) -> List[Tensor]:
+            """
+            This function return a list with the quantizer's quantization parameters variables.
```

Comment:
returns
parameters or variables?

Reply:
These are the variables that represent the quantization parameters.
I would have changed it but this method appears in 5 different locations with this same comment.
So unless you think it is totally confusing, I prefer to not change it to prevent messing up the PR.